ntoskrnl/mm/ARM3/pagfault.c

```diff
@@ -2446,19 +2249,20 @@ _WARN("Session space stuff is not implemented yet!")
             ASSERT(CurrentThread->ApcNeeded == 0);
 
             /* Drop the working set lock */
-            if (TrapInformation)
-                MiUnlockProcessWorkingSet(CurrentProcess, CurrentThread);
-            ASSERT(KeGetCurrentIrql() == OldIrql);
+            MiUnlockProcessWorkingSetForFault(CurrentProcess, CurrentThread, &WsSafe, &WsShared);
 
             /* Handle stack expansion */
-            return MiCheckForUserStackOverflow(Address, TrapInformation);
+            Status = MiCheckForUserStackOverflow(Address, TrapInformation);
+
+            MiLockProcessWorkingSetForFault(CurrentProcess, CurrentThread, WsSafe, WsShared);
+            return Status;
         }
 
         /* Did we get a prototype PTE back? */
         if (!ProtoPte)
         {
             /* Is this PTE actually part of the PDE-PTE self-mapping directory? */
-            if (PointerPde == MiAddressToPde(PTE_BASE))
+            if (MiAddressToPde(Address) == MiAddressToPde(PTE_BASE))
```

Comment:
This looks broken. Should probably be something like `if (MiIsPageTableAddress(Address))`. But like the above block, I think it's useless.

Reply:
Moved in the PDE specific code